### ISSUE TITLE
Using "rawPayload" results in payload being base64 encoded

### ISSUE THREAD

**Issue by alecor191** [dapr/dotnet-sdk#1015]
## TL;DR

I have an app that publishes a message to Dapr pubsub with `rawPayload = true`. The payload is a (Protobuf-encoded) byte array. A second app subscribes to the pubsub component and is also configured with `rawPayload = true`. The the subscribing app receives the message, it receives a Base64 encoded string representing the payload the publishing app posted. Both apps use Dapr .NET SDK.

Is there a way to prevent Dapr from modifying provided payload? (I would like to remove any unnecessary encoding/decoding from my payload for performance reasons).

## Details

For my test, I'm using the Redis container that gets setup as part of `dapr init` as my pubsub component backing store.

### Publishing App

I have a simple app using Dapr .NET SDK that publishes content as byte array to a Dapr pubsub component. Here the code:

```csharp
byte[] dto = ... // Protobuf-encoded content as byte[]

await _daprClient.PublishEventAsync(
    componentName,
    topicName,
    dto,
    new Dictionary<string, string>
    {
        ["rawPayload"] = "true"
    },
    cancellationToken);
```

After publishing a message, I notice in the Redis container that the payload I sent is already stored as Base64 string

```text
127.0.0.1:6379> dump pub-app-protobuf
"\x0f\x01\x10\x00\x00\x01\x85\xfd\x05\xff\x1d\x00\x00\x00\x00\x00\x00\x00\x00@\xa7\xa7\x00\x00\x00\n\x00\x01\x01\x00\x01\x01\x01\x84data\x05\x00\x01\x02\x01\x00\x01\x00\x01\xe0\x86\"Ck5oZWxsbyBmcm9tIHB1Yi1hcHAgb24gbG9jYWxob3N0IHdpdGggUHJvdG9idWYgbWVzc2FnZSAyMDIzLTAxLTI5VDEwOjE1OjA2LjU5OFoaDAiqjtmeBhCEjtjLAiIDAQID\"\x01\x88\x04\x01\xff\x01\x81\x00\x00\x01\x85\xfd\x05\xff\x1d\x00\x00\t\x008\x9b,;U\x0e\xa5E
```

I.e. encoding of the provided payload already took place (see `Ck5oZWxsbyBmcm9tI...`; that's the Base64-encoded payload the app sent). Is there a way to avoid this?

### Subscribing App

```csharp
[Topic(PubAppProtobufComponentName, PubAppProtobufTopicName, true)]
[Route("protobuf")]
[HttpPost]
public async Task<IActionResult> ProcessProtobufMessage()
{
    // Request.Body stream contains a string that is the Base64-encoded Protobuf-message.
```

Here I would have expected to get a byte[] back, as that's what the publisher posted. Is this expected that `rawPayload = true` on the subscriber-side means receiving a Base64-encoded payload? Or am I doing something wrong?

### Potentially related ticket

I observed similar behavior (i.e. binary content getting converted to Base64 string) also with the JS SDK (see https://github.com/dapr/js-sdk/issues/447) . I.e. if this is not a .NET SDK-specific issue, but rather something  I should post to `dapr` project, please let me know.

**Comment by alecor191** [comment]
Closing this issue as I believe it will be addressed once https://github.com/dapr/dapr/issues/4541 is done.
